java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -497,4 +497,28 @@ private static void addLearnedConstraint(ImmutableSet.Builder<LearnedConstraint>
     return result.build();
   }
 
+  /**
+   * To be used ONLY by ExplodedGraph viewer
+   */
+  @VisibleForTesting
```

Comment:
I tend to think that we may want to provide a class (in the tests) that will do the job of converting a PS to a String rather than adding an API in production code.